src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,26 @@
+import { BadgeProps } from './Badge.d';
+import './Badge.css';
+
+export const Badge = ({
+  name,
+  size = 24,
+  color = 'currentColor',
+  ...rest
+}: BadgeProps): JSX.Element => {
+  const icons = {
```

Comment:
store the icon in a separate file, as its own component, refer to Star Icon for example, you can use the same icon/component here.